### ISSUE TITLE
Smokey Stops reporting the first feedback for posts

### ISSUE THREAD

**Issue by SulphurDioxide** [Charcoal-SE/SmokeDetector#950]
I've seen this happen a few times. After a period of not being rebooted, smokey stops adding chat messages in CharcoalHQ like this: 
![fpglor](https://user-images.githubusercontent.com/14162920/28220555-da4ab486-68b7-11e7-88e8-d721ed491ca1.PNG)
Instead, not message appears even though people _have_ provided feedback for the post. 

Rebooting smokey seems to fix the problem. I think someone mentioned a websocket timing out when we have discuss this in the past but I can't be sure.

**Comment by j-f1** [comment]
This is why I think we should switch to a proper ActionCable client implementation.

**Comment by quartata** [comment]
Has this happened again recently?

**Comment by SulphurDioxide** [comment]
I'll keep an eye out for the next few days. It definitely happened within 2 weeks.

**Comment by angussidney** [comment]
Has this been reproduced recently?

**Comment by SulphurDioxide** [comment]
Nope. I haven't seen this recently.

**Comment by SulphurDioxide** [comment]
This has started happening again. See the transcript here: https://chat.stackexchange.com/transcript/message/45606961#45606961

**Comment by EliahKagan** [comment]
This has been happening [fairly often lately](https://chat.stackexchange.com/search?q=not+showing+first+feedback&room=11540), though I don't know for sure if it has increased in frequency. It doesn't seem to depend on which instance of Smokey is being used. Smokey isn't losing its ability to post messages in the room--it still reports posts and replies to commands.

@double-beep has observed that it seems to be triggered by a large number of commands issued in a short time.

**Comment by makyen** [comment]
We're going to need to watch to see if this is actually resolved. However, I fixed an issue which would have caused the problem.

The [issue I fixed](https://github.com/Charcoal-SE/SmokeDetector/commit/1365be9019ab91c5a8c9996c837a2fd6cec0f9ee) was that after the WebSocket to MS is initially established, if an exception occurred (e.g. an error or the WebSocket was closed by MS), then SD would attempt to re-establish the connection, but would not transmit it's security `key`, which would cause the attempt to reestablish the WebSocket to fail in a way that caused SD to think that things were OK. Thus, once disconnected, the WebSocket connection would never be actually reestablished.
